app/src/layouts/cards/components/card.vue

```diff
@@ -107,14 +115,51 @@ export default defineComponent({
 			if (props.file.type.startsWith('image') === false) return null;
 			if (props.file.type.includes('svg')) return null;
 
-			let key = 'system-medium-cover';
+			const fitParam = props.crop ? 'cover' : 'inside';
+			const url =
+				getRootPath() +
+				`assets/${props.file.id}?width=300&height=300&fit=${fitParam}&withoutEnlargement=true&modified=${props.file.modified_on}`;
+			return addTokenToURL(url);
+		});
 
-			if (props.crop === false) {
-				key = 'system-medium-contain';
+		watch(
+			() => props.file.id,
+			(newValue, oldValue) => {
+				if (newValue === oldValue) return;
+
+				if (newValue) {
+					fetchImgMeasurements();
+				}
+			},
+			{ immediate: true }
+		);
+
+		async function fetchImgMeasurements() {
+			try {
+				const response = await api.get(`/files/${props.file.id}`, {
+					params: {
+						fields: ['width', 'height'],
+					},
+				});
+				imgMeasurements.value = response.data.data;
```

Comment:
This is impossible unfortunately. We can't make a request for every file on the page, as you could end up doing 1000 requests on file library open 🤔